src/Options.php

```diff
@@ -648,6 +650,30 @@ public function getIntegrations()
         return $this->options['integrations'];
     }
 
+    public function setTransport(TransportInterface $transport): void
+    {
+        $options = array_merge($this->options, ['transport' => $transport]);
+
+        $this->options = $this->resolver->resolve($options);
+    }
+
+    public function getTransport(): ?TransportInterface
+    {
+        return $this->options['transport'];
+    }
+
+    public function setHttpClient(HttpClientInterface $httpClient): void
+    {
+        $options = array_merge($this->options, ['http_client' => $httpClient]);
+
+        $this->options = $this->resolver->resolve($options);
+    }
+
+    public function getHttpClient(): ?HttpClientInterface
+    {
+        return $this->options['http_client'];
+    }
```

Comment:
I'm wondering if we should only allow the `http_client` option. Since right now setting a `transport` overshadows the client. If you need a custom transport I don't think it's bad to use the client builder directly but setting the `http_client` seems like a reasonable thing to have especially since the `transport` has quite a few more dependencies needed to work in almost all cases, WDYT?

Reply:
I'm aware of some people who added custom retry logic and such. I would like to keep this as is, to ease swapping out the transport. The current logic in `ClientBuilder::__construct` should work well, or am I overlooking something?

Reply:
Moved to https://github.com/getsentry/sentry-php/issues/1595